### ISSUE TITLE
Building BUILDING_EDEN_MANSIONS has no landing pads - save files damaged

### ISSUE THREAD

**Issue by Quickmind01** [OpenApoc/OpenApoc#666]
After a crash in a UFO mission, I reloaded my save for both an in-mission file and out of mission file. Both saves give the "Building BUILDING_EDEN_MANSIONS has no landing pads" error and closes the game. This may be referring to a Government building missing its landing pad after I destroyed it trying to free up a path to a crashed construction vehicle. The game ran fine prior to reloading the game and save files.
[save_Stream Test - New Battle (2).zip](https://github.com/OpenApoc/OpenApoc/files/3438276/save_Stream.Test.-.New.Battle.2.zip)

**Comment by makus82** [comment]
and Q didnt mention that he use previously debug hotkeys for destroy terrain around...

**Comment by Quickmind01** [comment]
Correct. I used the debug keys to destroy the terrain at and around the landing pad.

**Comment by Quickmind01** [comment]
Attached is a save from a previous version (within the last week). I loaded this up and the landing pad is missing from the same building, but the game loads fine.

![Eden Building Landing Pad Bug](https://user-images.githubusercontent.com/51022165/61996037-85805680-b055-11e9-9810-0f07ea24e7dd.png)


[save_Stream Test 2.zip](https://github.com/OpenApoc/OpenApoc/files/3438283/save_Stream.Test.2.zip)

**Comment by JonnyH** [comment]
Yeah, checking for valid landing pads (as they "should" be invincible?) is part of the GameState validation done at save loading time.

I would argue that it should be a fatal error, as the game simply doesn't handle buildings that aren't accessible.

**Comment by Quickmind01** [comment]
The landing pads were destructible in vanilla, though. I remember
destroying them in an attempt to prevent craft from coming out, though I
can't remember if it was 100% effective.

**Comment by FilmBoy84** [comment]
Yes, landing pads were destructible in original game

From testing here's the effects damage to landing pads have (it also applies to garages, although in original game there is a bug that means garages are not removed visually, even though they are destroyed)

* 1) The building loses the ability to launch vehicles from the pad (or garage)
* 2) All vehicles in the landing pad/garage are destroyed
* 3) The cost of the vehicles in the landing pad/garage is deducted from the parent organisations money (If you take the time to tally up the parent organisations finances before destroying the pad, you will see the cost of the tile, plus a variable amount of cash according to what vehicles were in disappear from their coffers)
* 4) Any agents of the parent organisation in the landing pad/garage, for example those assigned to craft, are killed which reduces the organisations ability to field large numbers of agents against X-COM in say a base raid unless they have other buildings (This is all done back-end, there's no visual feedback... Skin36 may be best placed to say exactly what goes on)

The concerns with the above and OpenApoc is that i do not believe we currently store any information on vehicles held within organisation property.

I believe vehicles are spawned/destroyed according to mission generation. This does happen in original game, but there appears to be a holding time before/after each mission in which the vehicles are present in the landing pad/garage can be destroyed there also appears to be a template for what each organisation can launch from it's buildings against aggressors based on available finances.

Most certainly we need some information on how this is all done in the code for the original game. But it appears than OpenApoc currently falls drastically short of managing buildings, vehicles and organisations in the same manner as the original game.

**Comment by FilmBoy84** [comment]
Playing around with the original game more, i suspect the holding time for organisation vehicles in each of their buildings is reset from the start of each week where a review is done of what vehicles/equipment/agents/buildings it has access to and compared to X-COM score and week number for purposes of availability and tech level

Missions are then planned based on this review

The number of vehicles and agents available to each organisation changes according to the finances at any given time

If X-COM destroy the vehicles or agents required for a mission, and the organisation has enough funds to recruit or purchase replacements it shall do so, allowing missions to be generated according to it's weekly plan

If the parent organisation has had it's funds reduced by, for example, loss of property, destruction of vehicles, equipment or agents from raiding or conflict with other organisations, then it will attempt to purchase all it can and hold them within it's buildings

It appears, under rare circumstances, that if X-COM or another organisation attack a building, the parent organisation can also generate, in addition to craft to defend the property cargo and taxi craft that move equipment and personnel to other, safe, buildings. These can also be destroyed costing the parent organisation whatever lies within.

It appears the original game is exceptionally more complex than i first anticipated... But then ive never really taken time to observe, in detail, what is happening user-side for such encounters.

**Comment by Quickmind01** [comment]
Excellent work and observations! I wish I had looked at that stuff more in-depth when I played it in the past. I think OpenAPOC will get there in time.

**Comment by Quickmind01** [comment]
Here is a clip from my stream to show what was going on. Note towards the end of the clip that I load another save file with the same landing pad destroyed, but the game was able to load it fine.

https://www.twitch.tv/videos/459284152

**Comment by FilmBoy84** [comment]
When loading this game (created when testing the city repair function and destroying pads to prevent access - build 20231228-6 ) an error message complaining of no landing pads will be shown and then allow the user to continue loading

Buildings without landing pads are permissible in the original game so this warning should be silenced (though still logged)

We also need to check if destroying landing pads or garages in openapoc has any unintended consequences...

[save_Pre Night 1 Building Repair.zip](https://github.com/OpenApoc/OpenApoc/files/13809480/save_Pre.Night.1.Building.Repair.zip)
![image](https://github.com/OpenApoc/OpenApoc/assets/33913088/385620da-c832-41bb-bc98-38cfea9f7d09)
![image](https://github.com/OpenApoc/OpenApoc/assets/33913088/83ce162b-f7f5-4a6b-a830-fac49a64574b)
